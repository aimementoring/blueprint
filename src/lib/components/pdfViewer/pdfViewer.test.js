import React from 'react';
import renderer from 'react-test-renderer';
import PdfViewer from './pdfViewer.js';

describe('PdfViewer', () => {
  it('renders properly with all props used', () => {
    const tree = renderer
      .create(
        <PdfViewer
          pdf="https://archive.org/download/starrover00lond/starrover00lond.pdf"
          checkboxLabel="Testing my checkbox"
          onChange={value => console.log(value)}
          height={300}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly only with pdf value, without checkbox', () => {
    const tree = renderer
      .create(<PdfViewer pdf="https://archive.org/download/starrover00lond/starrover00lond.pdf" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly with pdf, and checkbox', () => {
    const tree = renderer
      .create(
        <PdfViewer
          pdf="https://archive.org/download/starrover00lond/starrover00lond.pdf"
          checkboxLabel="Testing my checkbox"
          onChange={value => console.log(value)}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
