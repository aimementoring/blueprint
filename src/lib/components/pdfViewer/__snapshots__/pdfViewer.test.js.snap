// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PdfViewer renders properly only with pdf value, without checkbox 1`] = `
<div
  className=""
>
  <div
    className="theme-portal"
  >
    <div
      className="pdfContainer "
      style={Object {}}
    >
      <object
        className="objectContainer"
        data="https://archive.org/download/starrover00lond/starrover00lond.pdf#view=FitH"
        type="application/pdf"
      >
        <embed
          src="https://archive.org/download/starrover00lond/starrover00lond.pdf#view=FitH"
          type="application/pdf"
        />
      </object>
    </div>
  </div>
</div>
`;

exports[`PdfViewer renders properly with all props used 1`] = `
<div
  className=""
>
  <div
    className="theme-portal"
  >
    <div
      className="pdfContainer "
      style={
        Object {
          "height": "300px",
        }
      }
    >
      <object
        className="objectContainer"
        data="https://archive.org/download/starrover00lond/starrover00lond.pdf#view=FitH"
        type="application/pdf"
      >
        <embed
          src="https://archive.org/download/starrover00lond/starrover00lond.pdf#view=FitH"
          type="application/pdf"
        />
      </object>
    </div>
    <div
      className="theme-purple wrapper"
    >
      <div
        className=" customCheckbox customCheckboxDefault"
        onClick={[Function]}
      >
        <input
          name="checkbox"
          type="hidden"
          value="no"
        />
        <input
          checked={false}
          className="false"
          name="checkbox"
          readOnly={true}
          type="checkbox"
          value="yes"
        />
        <label>
          Testing my checkbox
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`PdfViewer renders properly with pdf, and checkbox 1`] = `
<div
  className=""
>
  <div
    className="theme-portal"
  >
    <div
      className="pdfContainer "
      style={Object {}}
    >
      <object
        className="objectContainer"
        data="https://archive.org/download/starrover00lond/starrover00lond.pdf#view=FitH"
        type="application/pdf"
      >
        <embed
          src="https://archive.org/download/starrover00lond/starrover00lond.pdf#view=FitH"
          type="application/pdf"
        />
      </object>
    </div>
    <div
      className="theme-purple wrapper"
    >
      <div
        className=" customCheckbox customCheckboxDefault"
        onClick={[Function]}
      >
        <input
          name="checkbox"
          type="hidden"
          value="no"
        />
        <input
          checked={false}
          className="false"
          name="checkbox"
          readOnly={true}
          type="checkbox"
          value="yes"
        />
        <label>
          Testing my checkbox
        </label>
      </div>
    </div>
  </div>
</div>
`;
